### PR TITLE
Loki: Show label options for unwrap operation

### LINKED ISSUE #48535: Loki Query Builder: Unwrap
Unwrap with property suggestions based on results

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

In this PR, we are helping users by showing label options for `Unwrap` operation. We are:

- Introducing new editor component `<UnwrapParamEditor />` where we are using `<Select />` component
- In `<UnwrapParamEditor />` when clicking on `<Select />`, we trigger `loadUnwrapOptions` that does following:
  - Runs `getDataSamples` that returns DataFrame with log samples
  - Extracts objects with label-values from `labels` field
  - Takes first object and filters out labels that have value that us supported by unwrap - 1. number value, 2. golang duration value, 3. number + bytes
  - Takes the array of possible labels and checks if this label is in all sampled lines
  - If it is, it is offered as option
- User can input their own label, if they want

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/48535

**Special notes for your reviewer**:
To test the solution, I have used `tns/demo`. You can follow up this video (and click around/change things), with following query `sum by(job) (sum_over_time({compose_project="docker-compose"} | logfmt | status = "200"  | __error__=`` [$__interval]))`

https://user-images.githubusercontent.com/30407135/181020627-49f0e5e6-114f-4b35-8e8c-49bf7aca4b51.mov

### CODE REVIEW

File: public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  const [state, setState] = useState<{
+    options?: Array<SelectableValue<string>>;
+    isLoading?: boolean;
+  }>({});
+
+  return (
+    <Select
+      inputId={getOperationParamId(operationIndex, index)}
+      onOpenMenu={async () => {
+        if (datasource instanceof LokiDatasource) {
```
Comment (gabor) [inline public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx:32]: this is always `true`, right?

maybe we could add a comment here along the lines of:
`// this check is always true, we do it to make typescript happy` or something similar, to make it easier for the next person reading it understand why the `if`.

Comment (gabor) [inline public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx:32]: this is always `true`, right?

maybe we could add a comment here along the lines of:
`// this check is always true, we do it to make typescript happy` or something similar, to make it easier for the next person reading it understand why the `if`.

**Review by gabor — APPROVED** [review]
LGTM! i left a comment for a potential improvement, but it's not important.

File: public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/querybuilder/components/UnwrapParamEditor.tsx:33]: Good point. Added in https://github.com/grafana/grafana/pull/52810/commits/2908cd0b03e062d1b96cdd256181f81c31d3c0b3.

**Pull request merged**